premake5.lua

```diff
@@ -538,6 +538,11 @@ tool "gfx"
          removefiles { "tools/gfx/circular-resource-heap-d3d12.cpp", "tools/gfx/d3d-util.cpp", "tools/gfx/descriptor-heap-d3d12.cpp", "tools/gfx/render-d3d11.cpp", "tools/gfx/render-d3d12.cpp", "tools/gfx/render-gl.cpp", "tools/gfx/resource-d3d12.cpp", "tools/gfx/render-vk.cpp", "tools/gfx/vk-swap-chain.cpp", "tools/gfx/window.cpp" }
     end
 
+	-- Remove VK from OSX gfx build
```

Comment:
Note really relevant to this PR:

It seems like eventually we need to put the API- and platform-specific code into subdirectories so that rather than *removing* all the VK (or D3D) stuff on certain platforms, we can simply *add* the API- or platform-specific code where relevant.

E.g., if we had `tools/gfx/vk/*` instead of `tools/gfx/vk-*` or `tools/gfx/*-vk` this might be a lot cleaner.

Reply:
Yes, this would be significantly better.